### PR TITLE
Update to 20.0.0

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,8 +35,9 @@ test:
     - python -m unittest discover -p "test_storage_*.py"
 
 about:
-  home: https://github.com/Azure/azure-sdk-for-python
-  license: Apache-2.0
+  home: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-mgmt-storage
+  license: MIT
+  license_file: LICENSE
```
Comment (dhirschfeld) [inline recipe/meta.yaml:40]: https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-storage_20.0.0/sdk/storage/azure-mgmt-storage/LICENSE

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {% set name = "azure-mgmt-storage" %}
-{% set version = "6.0.0" %}
-{% set sha256 = "5cccfe6af0529a8a8a0557d130191a4a269fd2eac13ba6d6abd97ca851eef55d" %}
+{% set version = "20.0.0" %}
+{% set sha256 = "6ee476b5622ff6f5954edee6eb0d8dd427b3217022855adf1ec8633ca04cdee2" %}
```
Comment (dhirschfeld) [inline recipe/meta.yaml:3]: https://pypi.org/project/azure-mgmt-storage/20.0.0/

File: recipe/meta.yaml [hunk 3]
```diff
@@ -18,13 +18,13 @@ build:
...
+    - python >=3.6
     - pip
   run:
-    - python >=3
-    - azure-common >=1.1
-    - msrestazure >=0.4.11
+    - python >=3.6
+    - msrest >=0.6.21
+    - azure-common ~=1.1
+    - azure-mgmt-core >=1.3.0,<2.0.0
```
Comment (dhirschfeld) [inline recipe/meta.yaml:27]: https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-storage_20.0.0/sdk/storage/azure-mgmt-storage/setup.py#L68-L71

**Review by Maxyme — APPROVED** [review]

**Pull request merged**